spark/src/main/java/com/netflix/iceberg/spark/source/Writer.java

```diff
@@ -242,15 +252,19 @@ public String toString() {
     private final OutputFile file;
     private FileAppender<InternalRow> appender = null;
     private Metrics metrics = null;
+    private final EncryptionKeyMetadata keyMetadata;
 
     UnpartitionedWriter(
         OutputFile file,
         FileFormat format,
         AppenderFactory<InternalRow> factory,
-        FileIO fileIo) {
-      this.file = file;
+        FileIO fileIo,
+        EncryptionManager encryptionManager) {
       this.fileIo = fileIo;
-      this.appender = factory.newAppender(file, format);
+      EncryptedOutputFile encryptedOutput = encryptionManager.encrypt(file);
+      this.file = encryptedOutput.encryptingOutputFile();
+      this.keyMetadata = encryptedOutput.keyMetadata();
+      this.appender = factory.newAppender(this.file, format);
```

Comment:
Correctness requires using `this.file` instead of `file`. In that case, the input argument should be renamed to deconflict instead. That will prevent future accidents.